pkg/kcp/provider/azure/vpcpeering/peeringLocalCreate.go

```diff
@@ -45,20 +45,25 @@ func peeringLocalCreate(ctx context.Context, st composed.State) (error, context.
 			state.ObjAsVpcPeering().Status.State = string(cloudcontrolv1beta1.ErrorState)
 		}
 
+		condition := metav1.Condition{
+			Type:    cloudcontrolv1beta1.ConditionTypeError,
+			Status:  metav1.ConditionTrue,
+			Reason:  cloudcontrolv1beta1.ReasonFailedCreatingVpcPeeringConnection,
+			Message: message,
+		}
```

Comment:
Couple lines above this you are setting `.status.state` to warning or error.
Please reconsider if its necessary to set this condition to `ConditionTypeWarning` in a case when `.status.state` is set to Warning as well.